### ISSUE TITLE
Components like tablist items, header/footer have no correct data

### ISSUE THREAD

**Issue by andreasdc** [SpigotMC/BungeeCord#3631]
### Bungeecord version

This server is running BungeeCord version git:BungeeCord-Bootstrap:1.20-R0.3-SNAPSHOT:ffa011c:1804 by md_5

### Bungeecord plugins

-

### The bug

After the lastest updates components like tablist items, header, footer don't have their data like: bold, italic, striketrough. I suppose this commit is the fault https://github.com/SpigotMC/BungeeCord/commit/737d545fb60523bd7deafd5551f06615c0b87f0e.

### Checking

- [X] I am using BungeeCord and **not a fork**. Issues with forks should not be reported here.
- [X] I think this is **not** an issue with a bungeecord plugin.
- [ ] I have not read these checkboxes and therefore I just ticked them all.
- [X] This is not a question or plugin creation help request.

**Comment by bob7l** [comment]
Unless I'm misunderstanding, I can't reproduce this issue on 1.8.9 or 1.20.4 using build 1804. 

https://i.gyazo.com/4bf677b5469aae03f8debf4064f9bd8b.png

**Comment by andreasdc** [comment]
I was testing on 1.20.2 client, but I dont think thats the case. I was testing with older bungee and I didnt have it, I will test it again later.

**Comment by andreasdc** [comment]
They are sent by BungeeTabListPlus by bungee.

**Comment by Janmm14** [comment]
> They are sent by BungeeTabListPlus by bungee.

Maybe @bob7l is using different bungee plugin or using a spigot plugin?

Either way it'd be best if you could do simple bungeecord commit-by-commit testing.

Could you provide one example source string which is incorrectly un-bolded? Or is just a simple `§lAbcd` also getting un-bolded?

**Comment by andreasdc** [comment]
> > They are sent by BungeeTabListPlus by bungee.
> 
> Maybe @bob7l is using different bungee plugin or using a spigot plugin?
> 
> Either way it'd be best if you could do check whether that commit is actually the problem.
> 
> Could you provide one example source string which is incorrectly un-bolded? Or is just a simple `§lAbcd` also getting un-bolded?

All strings are unbolded and don't have strikethrough. Let me try again.

**Comment by andreasdc** [comment]
This commit broke it...
https://github.com/SpigotMC/BungeeCord/commit/231024ba42851d991cf5ad7f18633b57dab07cbd

**Comment by md-5** [comment]
Pretty sure that commit fixed an issue basically the same as what this ticket describes

**Comment by andreasdc** [comment]
This build works https://ci.md-5.net/job/BungeeCord/1776/
and this doesn't work https://ci.md-5.net/job/BungeeCord/1777/
Earlier I didn't have any issues with objects components, maybe it's a small mistake or something.

**Comment by bob7l** [comment]
Anyone else able to reproduce this?

I just tried BungeeTabListPlus on Bungee alone, and even BungeeTabListPlus on Spigot alone and still can't reproduce it. Used a combination of hex colors, strikethrough, bold, etc. Works fine. 

@andreasdc Maybe you should add some debug code in Bungee to outprint the JSON string before it's deserialized so we can see exactly what's being read

**Comment by andreasdc** [comment]
I think it get's to check '"true"', thjen it doesn't see it as boolean. Other objects are just 'true' and they are boolean.

**Comment by md-5** [comment]
I guess .getAsBoolean() used to parse string to Boolean.

I think whatever is writing a boolean as a string is wrong, but bungee could be more relaxed I guess

**Comment by Janmm14** [comment]
@andreasdc I think your BungeeTablistPlus version is from before jan 20 2024, because thats where they fixed this: https://github.com/CodeCrafter47/TabOverlayCommon/commit/af983094c422ba7410cc3d322b7e8eb496691638

**Comment by andreasdc** [comment]
Yup, I have the old version, didn't see that update. Thanks for your finding!
